src/librustdoc/clean/mod.rs

```diff
@@ -327,17 +327,27 @@ impl Item {
         }
     }
 
-    pub fn stability_class(&self) -> String {
-        self.stability.as_ref().map(|ref s| {
-            let mut base = match s.level {
-                stability::Unstable => "unstable".to_string(),
-                stability::Stable => String::new(),
-            };
-            if !s.deprecated_since.is_empty() {
-                base.push_str(" deprecated");
+    pub fn stability_class(&self) -> Option<String> {
+        match self.stability {
```

Comment:
Ah, yes, you're right. It was originally written that way, but I changed it because I got confused between `Option::map` and `Option::and_then`, thinking the closure couldn't return an option. I'll switch it back.